### PR TITLE
[avocado] Upgrade avocado to 103.X LTS

### PR DESCRIPTION
**Pull request by arif-ali**
Extra changes required due to module and function changes

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?
- [ ] Are all passwords or private data gathered by this PR [obfuscated](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin#how-to-prevent-collecting-passwords)?

### CODE REVIEW

**Comment by arif-ali** [comment]
Just the initial PR to see cirrus CI play and work with 103.X . The tox stuff will not affect this, so will add those in once that PR merges

**Comment by arif-ali** [comment]
I'm intrigued, does `cups` service not exist on centos stream 9 by default, and why this hasn't been picked up before 🤔 

Can one of the RH folks check this please?

**Comment by pmoravec** [comment]
Interesting.. I do see `cups`  installed by default on RHEL9 (I expect I used a default installation), but I do not see it installed on the tested CentOS 9 system - downloading the logs and unpacking generated sosreports, I see e.g.:

    $ grep cups sosreport-cirrus-task-5893501009788928-2024-04-10-todonci/installed-rpms
    $ 

(two sosreports contain `cups-libs` but not `cups` itself).

So we would need to either install `cups` (prior testing on CentOS 9) or use some different plugin/service?

**Comment by jcastill** [comment]
But we've never seen this failure before, right? If that's the case, could it be a problem with the image itself?

**Comment by arif-ali** [comment]
Yeah I know that we've not seen it before, hence my curiosity on this.

The image is the same, as we've not changed it here in the CI, so not sure what is going on.

I've had 2 runs with the same result. Could be a result of how the new avocado is working and the old one missed it 🤷🏽

**Comment by jcastill** [comment]
I have a centos stream with the same version a we have here, so lets see if I can reproduce the issue.

**Comment by jcastill** [comment]
No luck, the tests in the stream image worked perfectly:

```
 (53/91) tests/report_tests/plugin_tests/defaults.py:DefaultCollectionsTest.test_service_status_collected:  PASS (45.73 s)
 (54/91) tests/report_tests/plugin_tests/defaults.py:DefaultCollectionsTest.test_journal_collected:  PASS (0.01 s)

```
Checking the previous results, for the tox.init PR, the test was successful as well:

```
 (53/91) tests/report_tests/plugin_tests/defaults.py:DefaultCollectionsTest.test_service_status_collected:  PASS (28.53 s)
 (54/91) tests/report_tests/plugin_tests/defaults.py:DefaultCollectionsTest.test_journal_collected:  PASS (0.00 s)
```

That said, in this case the tests were increased by 8 (result of #3591 ?) so the numbers don't match:

 ```
(57/99) tests/report_tests/plugin_tests/defaults.py:DefaultCollectionsTest.test_service_status_collected: STARTED
 (57/99) tests/report_tests/plugin_tests/defaults.py:DefaultCollectionsTest.test_service_status_collected:  FAIL: /var/tmp/avocado_lno96b1qsos_tests.py.DefaultCollectionsTest/sosreport-DefaultCollectionsTest/sosreport-cirrus-task-5893501009788928-2024-04-10-upbcuac/sos_commands/cups/systemctl_status_cups does not exist (4.85 s)
 (58/99) tests/report_tests/plugin_tests/defaults.py:DefaultCollectionsTest.test_journal_collected: STARTED
 (58/99) tests/report_tests/plugin_tests/defaults.py:DefaultCollectionsTest.test_journal_collected:  FAIL: /var/tmp/avocado_lno96b1qsos_tests.py.DefaultCollectionsTest/sosreport-DefaultCollectionsTest/sosreport-cirrus-task-5893501009788928-2024-04-10-upbcuac/sos_commands/cups/journalctl_--no-pager_--unit_cups does not exist (0.04 s)
```
I'm still getting familiar with the avocado tests, but could a previous test remove the cups rpm somehow?

**Comment by jcastill** [comment]
Interesting development - I managed to reproduce in Centos Stream 9:

```
 (57/99) tests/report_tests/plugin_tests/defaults.py:DefaultCollectionsTest.test_service_status_collected: FAIL: /var/tmp/avocado_nwpdtn1ysos_tests.py.DefaultCollectionsTest/sosreport-DefaultCollectionsTest/sosreport-localhost-2024-04-11-hewiakv/sos_commands/cups/systemctl_status_cups does not exist (25.62 s)
 (58/99) tests/report_tests/plugin_tests/defaults.py:DefaultCollectionsTest.test_journal_collected: STARTED
 (58/99) tests/report_tests/plugin_tests/defaults.py:DefaultCollectionsTest.test_journal_collected: FAIL: /var/tmp/avocado_nwpdtn1ysos_tests.py.DefaultCollectionsTest/sosreport-DefaultCollectionsTest/sosreport-localhost-2024-04-11-hewiakv/sos_commands/cups/journalctl_--no-pager_--unit_cups does not exist (0.02 s)
```

What makes this more interesting is that the service is there, running:
```

# systemctl status cups
● cups.service - CUPS Scheduler
     Loaded: loaded (/usr/lib/systemd/system/cups.service; enabled; preset: enabled)
    Drop-In: /usr/lib/systemd/system/cups.service.d
             └─server.conf
     Active: active (running) since Thu 2024-04-11 13:30:17 BST; 19min ago
TriggeredBy: ● cups.path
             ● cups.socket
       Docs: man:cupsd(8)
   Main PID: 867 (cupsd)
     Status: "Scheduler is running..."
      Tasks: 1 (limit: 7738)
     Memory: 3.3M
        CPU: 14ms
     CGroup: /system.slice/cups.service
             └─867 /usr/sbin/cupsd -l

Apr 11 13:30:17 localhost.localdomain systemd[1]: Starting CUPS Scheduler...
Apr 11 13:30:17 localhost.localdomain systemd[1]: Started CUPS Scheduler.
```

So I can go more in depth now and see what's going on. Unless anybody else has any other ideas, of course.

**Comment by arif-ali** [comment]
> Interesting development - I managed to reproduce in Centos Stream 9:
> 
> ```
>  (57/99) tests/report_tests/plugin_tests/defaults.py:DefaultCollectionsTest.test_service_status_collected: FAIL: /var/tmp/avocado_nwpdtn1ysos_tests.py.DefaultCollectionsTest/sosreport-DefaultCollectionsTest/sosreport-localhost-2024-04-11-hewiakv/sos_commands/cups/systemctl_status_cups does not exist (25.62 s)
>  (58/99) tests/report_tests/plugin_tests/defaults.py:DefaultCollectionsTest.test_journal_collected: STARTED
>  (58/99) tests/report_tests/plugin_tests/defaults.py:DefaultCollectionsTest.test_journal_collected: FAIL: /var/tmp/avocado_nwpdtn1ysos_tests.py.DefaultCollectionsTest/sosreport-DefaultCollectionsTest/sosreport-localhost-2024-04-11-hewiakv/sos_commands/cups/journalctl_--no-pager_--unit_cups does not exist (0.02 s)
> ```
> 
> What makes this more interesting is that the service is there, running:
> 
> ```
> 
> # systemctl status cups
> ● cups.service - CUPS Scheduler
>      Loaded: loaded (/usr/lib/systemd/system/cups.service; enabled; preset: enabled)
>     Drop-In: /usr/lib/systemd/system/cups.service.d
>              └─server.conf
>      Active: active (running) since Thu 2024-04-11 13:30:17 BST; 19min ago
> TriggeredBy: ● cups.path
>              ● cups.socket
>        Docs: man:cupsd(8)
>    Main PID: 867 (cupsd)
>      Status: "Scheduler is running..."
>       Tasks: 1 (limit: 7738)
>      Memory: 3.3M
>         CPU: 14ms
>      CGroup: /system.slice/cups.service
>              └─867 /usr/sbin/cupsd -l
> 
> Apr 11 13:30:17 localhost.localdomain systemd[1]: Starting CUPS Scheduler...
> Apr 11 13:30:17 localhost.localdomain systemd[1]: Started CUPS Scheduler.
> ```
> 
> So I can go more in depth now and see what's going on. Unless anybody else has any other ideas, of course.

That would be awesome, I've reproduced it in a GCE env too (but that was a personal account). So you have the means to do the extra testing on this, then that would be awesome.

it's bizarre that the issue is on centos stream 9 and not 8, so something with the combination of 103.0 and and 9 that is not quite there (plus my RPM/DNF and how EL works and differences with ubuntu is a bit rusty these days)

**Comment by jcastill** [comment]
On the other hand, the tests for the kafka plugin for centos stream 9 stage 2  went through without problems:

https://cirrus-ci.com/task/5246215078543360

**Comment by arif-ali** [comment]
> On the other hand, the tests for the kafka plugin for centos stream 9 stage 2 went through without problems:
> 
> https://cirrus-ci.com/task/5246215078543360

Yeah, but that is still using avocado 94. This seems like an issue with the way avocado 103.0 is potentially handling it.

**Comment by arif-ali** [comment]
tested this latest patch in my local env, and it worked. Hopefully this will be good to go 🤞🏽 

avocado now sees centos stream 9 as `centos-stream` and not `centos` and that is what caused the issue and we just map the packages for `rhel` to `centos-stream` as well

**Comment by TurboTurtle** [comment]
Beat me to it, but yep I was going to say the package map for distro packages. Good find!

CI is looking good, and I think we can move this out of draft.

File: .cirrus.yml [hunk 1]
```diff
@@ -134,7 +134,7 @@ deb_build_task:
         apt -y install devscripts equivs python3-pip
         mk-build-deps
         apt -y install ./sosreport-build-deps*.deb
-        pip3 install avocado-framework==94.0 --break-system-packages
+        pip3 install "avocado-framework<104.0" --break-system-packages
```
Comment (dnegreira) [inline .cirrus.yml:137]: Shouldn´t we want to explicitly pin it to the 103 LTS, IE:
`avocado-framework==103` ?

Comment (arif-ali) [reply .cirrus.yml:137]: based on the documentation from avocado-framework, they suggest to use this format to always pick the latest LTS. So, when 103.1 comes out, this should do it. There is no 103, but 103.0. I hope that makes sense

Comment (dnegreira) [reply .cirrus.yml:137]: Sounds good :+1:

File: tests/sos_tests.py [hunk 2]
```diff
@@ -819,7 +820,7 @@ def setUp(self):
         # seems awkward, but check_installed() and remove() are not exposed
         # together with install_distro_packages()
         self.installer = software_manager
-        self.sm = self.installer.SoftwareManager()
+        self.sm = self.installer.manager.SoftwareManager()
```
Comment (TurboTurtle) [inline tests/sos_tests.py:823]: With the changes between `SoftwareManager()` and `distro_packages` that we've found, we can un-awkward this bit and clean it up, e.g. simply:

```python
self.sm = software_manager.manager.SoftwareManager()
```

And drop the notes on lines 819+

**Review by TurboTurtle — APPROVED** [review]

**Review by pmoravec — APPROVED** [review]

**Pull request merged**